test/recipes/70-test_sslcbcpadding.t

```diff
@@ -0,0 +1,110 @@
+#! /usr/bin/env perl
+# Copyright 2016 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the OpenSSL license (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+
+use strict;
+use OpenSSL::Test qw/:DEFAULT cmdstr srctop_file bldtop_dir/;
+use OpenSSL::Test::Utils;
+use TLSProxy::Proxy;
+
+my $test_name = "test_sslcbcpadding";
+setup($test_name);
+
+plan skip_all => "TLSProxy isn't usable on $^O"
+    if $^O =~ /^(VMS|MSWin32)$/;
+
+plan skip_all => "$test_name needs the dynamic engine feature enabled"
+    if disabled("engine") || disabled("dynamic-engine");
+
+plan skip_all => "$test_name needs the sock feature enabled"
+    if disabled("sock");
+
+plan skip_all => "$test_name needs TLSv1.2 enabled"
+    if disabled("tls1_2");
+
+$ENV{OPENSSL_ia32cap} = '~0x200000200000000';
```

Comment:
It is needed. Ironically I think it isn't needed for most of the other tests, but for this one it is (try it without on an AES NI capable machine and you will get a "decryption failed or bad record mac" failure in the test). I think the boiler plate comes from the time when I first developed the Proxy and encountered this problem myself. The reason this one needs it is that it deletes the ETM extension. This means that OpenSSL will use one of "special" encrypt+mac ciphers. For reasons that I don't entirely understand and haven't really explored too closely, OpenSSL will prioritise the AES NI implementation over the engine implementation (I suspect a bug somewhere). By switching off AES NI support, the use of the ossltest engine is forced.